Add an activity for benchmarking only

kirklandsign opened the pull request:
Example usage: 
```
adb shell am start -n com.example.executorchllamademo/com.example.executorchllamademo.Benchmarking --es "model_path" "/data/local/tmp/llama/stories_kv_sdpa_fp32_xnn.pte" --es "tokenizer_path" "/data/local/tmp/llama/tokenizer.bin"
```

Then 
```
adb shell run-as com.example.executorchllamademo cat files/benchmark_results.txt
```

See result like
```
loadStart: 1722275116708
loadEnd: 1722275117629
generateStart: 1722275117629
generateEnd: 1722275118834
tokens/second: 105.445114
```

Note: We use activity because we assume it has higher RAM priority than a background service.

guangy10 commented on examples/demo-apps/android/LlamaDemo/app/src/main/AndroidManifest.xml:
```diff
@@ -47,6 +47,15 @@
+            android:name=".Benchmarking"
+            android:exported="true">
+            <intent-filter>
+                <action android:name="com.example.executorchllamademo.BENCHMARK" />
```
1. Can we name it to be something more generic, e.g. llm benchmark runner
2. Can we later move the entire app under `executorch/extension/llm` as an extension for llm benchmarking?

guangy10 replied:
2) is not a blocker for this PR. Once you addressed 1) this PR should be ready to go

kirklandsign replied:
Fixed 1 now. Working on moving it out of llamademoapp and use a separate app (for generic as well)

guangy10 commented on examples/demo-apps/android/LlamaDemo/app/src/main/java/com/example/executorchllamademo/Benchmarking.java:
```diff
@@ -0,0 +1,111 @@
+  long loadEnd;
+  long generateStart;
+  long generateEnd;
+  String tokens;
```
We would want to dump it to a standard and portable format later, e.g. json. Something we can reuse from AIBench.

kirklandsign replied:
That's a good idea and I plan to do that on the API as well

guangy10 replied:
There is a task for it [T197322159](https://www.internalfb.com/intern/tasks/?t=197322159). You can coordinate with Varun on it.

guangy10 commented:
In a follow up PR, you may want to connect this new apk to android-perf.yml here https://github.com/pytorch/executorch/blob/main/.github/workflows/android-perf.yml#L160-L162 and see if the test-spec could recognize it.

kirklandsign committed:
Add an activity for benchmarking only (#4443)

Summary:
Example usage: 
```
adb shell am start -n com.example.executorchllamademo/com.example.executorchllamademo.LlmBenchmarkRunner --es "model_path" "/data/local/tmp/llama/stories_kv_sdpa_fp32_xnn.pte" --es "tokenizer_path" "/data/local/tmp/llama/tokenizer.bin"
```

Then 
```
adb shell run-as com.example.executorchllamademo cat files/benchmark_results.txt
```

See result like
```
loadStart: 1722275116708
loadEnd: 1722275117629
generateStart: 1722275117629
generateEnd: 1722275118834
tokens/second: 105.445114
```

Note: We use activity because we assume it has higher RAM priority than a background service.


Differential Revision: D60399589

Pulled By: kirklandsign
